docs/DEVELOPING.md

```diff
@@ -88,16 +89,32 @@ sources:
 
 And that source will be available in Kibana.
 
+## Features
+
+Data source features will be populated to connector clients and customized connectors, to tell what features are available for the connector.
+
+The features are defined as class variables of the data source class, and the available features and default values are:
+1. `basic_rules_enabled` (Default: `True`) - whether basic sync rules are enabled
+2. `advanced_rules_enabled` (Default: `False`) - whether advanced rules are enabled
+3. `dls_enabled` (Default: `False`) - whether document-level security is enabled
+4. `incremental_sync_enabled` (Default: `False`) - whether incremental sync is enabled
+
+If you want to enable/disable any feature which is different from the default value, you need to override the class variable in the data source implementation. E.g. to enable incremental sync:
+```python
+class MyDataSource(BaseDataSource):
+    incremental_sync_enabled = True
+```
+
 ## Syncing
 ### Sync strategy
 
-In Elastic `connectors-python` a **Full sync** ensures full data parity (including deletion).
+There are two types of content sync jobs: full sync and incremental sync.
 
-This sync strategy is good enough for some sources like MongoDB where 100,000 documents can be fully synced in less than 30 seconds.
+A **Full sync** ensures full data parity (including deletion), this sync strategy is good enough for some sources like MongoDB where 100,000 documents can be fully synced in less than 30 seconds.
 
-As new sources are added, more sophisticated syncs may be introduced in order to achieve the same level of freshness available in Workplace Search.
+An **Incremental sync** only syncs documents created, updated and deleted since the last successful content sync.
```

Comment:
Might be worth to mention, that not every connector supports it and we don't support streaming updates (like binlog)